Core/Frameworks/Flake/Framework.php

```diff
@@ -261,6 +261,8 @@ protected static function initDb() {
         }
         if ($config['database']['mysql'] === true) {
             self::initDbMysql($config);
+        } elseif ($config['database']['pgsql'] === true) {
```

Comment:
I am getting a warning here. I do not get a warning for the mysql line above, even though it is set up to use sqlite. So something must be different between mysql and pgsql

> Warning: Undefined array key "pgsql" in /mnt/Entwicklung/htdocs/baikal/Core/Frameworks/Flake/Framework.php on line 264

Reply:
True, thanks for pointing that out! The issue was that the new `"pgsql"` config value was not defined in [baikal.yaml.dist](https://github.com/sabre-io/Baikal/blob/master/config/baikal.yaml.dist).

In order to address your 3rd comment (using a dropdown for the database backend on the settings page) I've replaced the `mysql` and `pgsql` config values with a combined `backend` value in the `database` config namespace.

These changes are present in 4ab75f76

Reply:
Doesn't this mean that existing users who upgrade without touching their yaml file would get the warning as well?

Reply:
Yes, but the mechanism was replaced with a combined variable for the database backend. However this is basically a breaking change as the `backend` setting won't exist in the yaml config of existing installations that upgrade to the new version.

A solution would be to implement a fallback to the previous `mysql` setting and throw a warning if the new `backend` setting is missing from the yaml. Should I implement that?

Reply:
I think a better solution would be to automatically update the config format when upgrading Baikal. We already perform migration steps somewhere

Reply:
Okay yeah, good idea! Implemented in ef019bc ([Compare](https://github.com/sabre-io/Baikal/compare/b322aaa08196797407137a68149d3c6bd931dcd0..ef019bcfd407ca32f6845d9384e1919bd4d772bc))